### PR TITLE
ENG-4983 Amend the ComponentManager endpoint for delete POST /components/{bundleCode}/uninstall

### PR DESCRIPTION
**Pull request by entgigi**

### CODE REVIEW

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java [hunk 1]
```diff
@@ -208,4 +228,34 @@ private JobResult executeUninstall(Installable<?> installable) {
...
+                log.error("with response:'{}' we had a json error", response, ex);
+            }
+            return JobStatus.UNINSTALL_PARTIAL_COMPLETED;
+
+        } else {
+            return JobStatus.UNINSTALL_COMPLETED;
+        }
+    }
+
+    private EntandoCoreComponentDeleteRequest convertEntityToDto(EntandoBundleComponentJobEntity entity) {
```
Comment (mirkoPan) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java]: Consider putting this method in the dto in order to have greater uniformity in the mapping process than solutions adopted elsewhere in the code. On the other hand, the coupling of this suggestion, which would bind the entity with the dto, must be considered,

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java [hunk 2]
```diff
@@ -208,4 +228,34 @@ private JobResult executeUninstall(Installable<?> installable) {
...
+    private JobStatus executeDeleteFromAppEngine(List<EntandoBundleComponentJobEntity> toDelete,
+            EntandoBundleJobEntity parentJob) {
+        EntandoCoreComponentDeleteResponse response = entandoCoreClient.deleteComponents(toDelete.stream()
+                .map(this::convertEntityToDto)
+                .collect(Collectors.toList()));
+
+        if (!EntandoCoreComponentDeleteResponseStatus.SUCCESS.equals(response.getStatus())) {
+            // FIXME what to do with failure status ???
+            try {
+                String uninstallErrors = om.writeValueAsString(response.getComponents().stream()
```
Comment (mirkoPan) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java]: consider using a more descriptive variable

Comment (entgigi) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java]: done, pls resolve if it's ok for you

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java (hunk 1 above)
Comment (entgigi) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java]: done, pls resolve if it's ok for you

**Review by mirkoPan — APPROVED** [review]
LGTM

File: src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java [hunk 3]
```diff
@@ -632,6 +636,47 @@ public EntandoCoreComponentUsage getComponentUsage(String code, String[] endpoin
...
+            Supplier<ResponseEntity<SimpleRestResponse<EntandoCoreComponentDeleteResponse>>> r = () ->
+                    restTemplate
+                            .exchange(
+                                    resolvePathSegments(API_PATH_SEGMENT, COMPONENTS_PATH_SEGMENT,
+                                            DELETE_ALL_INTERNALS_PATH_SEGMENT).build().toUri(),
+                                    HttpMethod.DELETE, new HttpEntity<>(request),
+                                    new ParameterizedTypeReference<SimpleRestResponse<EntandoCoreComponentDeleteResponse>>() {
+                                    });
+
+            var usage = (ResponseEntity<SimpleRestResponse<EntandoCoreComponentDeleteResponse>>) retryer.execute(r);
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java]: Are we deleting data or fetching usage details? Why is the var called `usage`?

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java [hunk 4]
```diff
@@ -127,9 +136,10 @@ private void submitUninstallAsync(EntandoBundleJobEntity parentJob, EntandoBundl
 
                 scheduler.queuePrimaryComponents(uninstallJobs);
 
-                JobProgress uninstallProgress = new JobProgress(1.0 / uninstallJobs.size());
+                JobProgress uninstallProgress = new JobProgress(1.0 / (uninstallJobs.size() + 1));
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java:147]: Please add a comment to explain why we need the new `+ 1`

File: src/test/java/org/entando/kubernetes/utils/TestInstallUtils.java [hunk 5]
```diff
@@ -1132,21 +1143,17 @@ public static String simulateInProgressUninstall(MockMvc mockMvc, EntandoCoreCli
         UniformDistribution delayDistribution = new UniformDistribution(200, 500);
         Mockito.reset(coreClient);
         WireMock.reset();
-        WireMock.setGlobalRandomDelay(delayDistribution);
+        //WireMock.setGlobalRandomDelay(delayDistribution);
```
Comment (firegloves) [inline src/test/java/org/entando/kubernetes/utils/TestInstallUtils.java]: please delete this commented line

File: src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java [hunk 6]
```diff
@@ -325,56 +324,42 @@ public void shouldCallCoreToUninstallComponents() throws Exception {
...
-        ac = ArgumentCaptor.forClass(String.class);
-        verify(coreClient, times(2)).setPageStatus(ac.capture(), ac.capture());
-        assertThat(ac.getAllValues()).containsAll(Arrays.asList("my-page", "another-page"));
-        assertThat(ac.getAllValues()).contains("draft");
-
-        ac = ArgumentCaptor.forClass(String.class);
-        verify(coreClient, times(2)).deletePage(ac.capture());
-        assertThat(ac.getAllValues()).containsAll(Arrays.asList("my-page", "another-page"));
-
+        */
```
Comment (firegloves) [inline src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java]: please delete these commented lines

**Review by firegloves — CHANGES_REQUESTED** [review]

File: src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java (hunk 3 above)
Comment (entgigi) [reply src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java]: done

File: src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java (hunk 4 above)
Comment (entgigi) [reply src/main/java/org/entando/kubernetes/service/digitalexchange/job/EntandoBundleUninstallService.java:147]: done

File: src/test/java/org/entando/kubernetes/utils/TestInstallUtils.java (hunk 5 above)
Comment (entgigi) [reply src/test/java/org/entando/kubernetes/utils/TestInstallUtils.java]: done

File: src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java (hunk 6 above)
Comment (entgigi) [reply src/test/java/org/entando/kubernetes/controller/mockmvc/InstallFlowTest.java]: done

**Review by firegloves — APPROVED** [review]
LGTM

**Pull request merged**